src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -264,6 +276,68 @@ public void Write(byte[] data)
 
         #region Private Methods
 
+        private TcpClient CreateClientFromNetworkInterface()
+        {
+            NetworkInterface ni;
+            IPAddress addr, hostAddr;
+            bool isRemoteHostIP = false;
+
+            try
+            {
+                //Input NI can be IPAddress string
+                if (IPAddress.TryParse(_options.NetworkInterface, out addr)) {
+                    var localEndPoint = new IPEndPoint(addr, 0);
+                    return new TcpClient(localEndPoint);
+                } else {
+                    //Get Network Interface
+                    ni = NetworkInterface.GetAllNetworkInterfaces().FirstOrDefault(x => x.Name == _options.NetworkInterface
+                        && x.OperationalStatus == OperationalStatus.Up);
+                }
+                
+                //Throw if input network adapter does not exist
+                if (ni == null) {
+                    WriteLog.To.Sync.I(Tag, $"Unknown Network Interface {_options.NetworkInterface}.");
+                    throw new CouchbaseNetworkException(C4NetworkErrorCode.UnknownHost);
+                }
+
+                //Get UnicastIPAddressInformationCollection from the NI adapter
+                var addresses = ni.GetIPProperties().UnicastAddresses;
+
+                // If remote host is IP, use the address family from the IP.
+                isRemoteHostIP = IPAddress.TryParse(_logic.UrlRequest.Host, out hostAddr);
+            
+                if (isRemoteHostIP) {
+                    return GetTcpClient(hostAddr.AddressFamily, addresses);
+                }
+
+                if (ni.Supports(NetworkInterfaceComponent.IPv6)) {
+                    return GetTcpClient(AddressFamily.InterNetworkV6, addresses);
+                } else if (ni.Supports(NetworkInterfaceComponent.IPv4)) {
+                    return GetTcpClient(AddressFamily.InterNetwork, addresses);
+                }
+
+                WriteLog.To.Sync.I(Tag, $"Network Interface {_options.NetworkInterface} supports neither IPv6 nor IPv4.");
+                throw new CouchbaseNetworkException(C4NetworkErrorCode.UnknownHost);
+
+            } catch (Exception ex) {
+                throw ex;
```

Comment:
This part does nothing except what would happen without the catch.  Either make it do something else or just don't include the try catch.